iothub/device/src/Transport/Mqtt/MqttTransportHandler.cs

```diff
@@ -716,22 +759,31 @@ private async Task UnsubscribeAsync(string topic, CancellationToken cancellation
                     .WithTopicFilter(fullTopic)
                     .Build();
 
-            MqttClientUnsubscribeResult unsubscribeResults = await _mqttClient.UnsubscribeAsync(unsubscribeOptions, cancellationToken).ConfigureAwait(false);
-
-            if (unsubscribeResults?.Items == null || unsubscribeResults.Items.Count != 1)
+            try
             {
-                throw new IotHubClientException($"Failed to unsubscribe to topic {fullTopic}", true);
-            }
+                MqttClientUnsubscribeResult unsubscribeResults = await _mqttClient.UnsubscribeAsync(unsubscribeOptions, cancellationToken).ConfigureAwait(false);
 
-            MqttClientUnsubscribeResultItem unsubscribeResult = unsubscribeResults.Items.FirstOrDefault();
-            if (!unsubscribeResult.TopicFilter.Equals(fullTopic))
-            {
-                throw new IotHubClientException($"Received unexpected unsubscription from topic {unsubscribeResult.TopicFilter}", true);
-            }
+                if (unsubscribeResults?.Items == null || unsubscribeResults.Items.Count != 1)
+                {
+                    throw new IotHubClientException($"Failed to unsubscribe to topic {fullTopic}", true);
```

Comment:
Can we set a StatusCode? Users need to be able to distinguish.

Reply:
This is more indicative of an MQTTNet bug than anything, so we don't have a status code for a case like this.

Reply:
Can we associate it with an exising one? Timeout?

Reply:
Maybe network error or server error. The only times I'd expect this exception to be thrown are if MQTTNet has a bug and returns a null/empty unsubscribeResults object (no status code for this case), if the hub acknowledges the incorrect unsubscriptions (server error), or if the packets sent over the wire experienced some loss and the ack sent by the service comes in malformed (network error).

Reply:
Of those scenarios, the only one I'd expect we'd ever see is packet loss messing with this field's results, so I'm leaning towards network errors as the status code

Reply:
Let's make a best match.

Reply:
I'll address this point in #2743 since I'm updating the ```isTransient``` flag on these same exceptions in that PR